src/community/smart-data-loader/src/main/java/org/geoserver/smartdataloader/data/SmartDataLoaderDataAccessFactory.java

```diff
@@ -284,9 +286,25 @@ private File saveMappingDocument(String pathname, String filename, Document mapp
     /** Method that allows to get a DataStoreInfo based on a set of parameters. */
     private DataStoreInfo getDataStoreInfo(Map<String, Serializable> params) throws IOException {
         String jdbcDataStoreId = lookup(DATASTORE_METADATA, params, String.class);
-        Catalog c = getGeoServer().getCatalog();
-        DataStoreInfo ds = c.getDataStore(jdbcDataStoreId);
-        return ds;
+        Catalog catalog = getGeoServer().getCatalog();
+        return catalog.getDataStore(jdbcDataStoreId);
+    }
+
+    static Map<String, Serializable> resolveParams(Map<String, Serializable> params) {
```

Comment:
Good intention but done in the wrong place IMHO, stores should not be aware of environment parameterization (or we'd have to make the all aware).

The GeoServer resource pool is [already resolving environment parameters on its own here](https://github.com/geoserver/geoserver/blob/99ae245402fca51e306eaf9817faaa891bf89e73/src/main/src/main/java/org/geoserver/catalog/ResourcePool.java#L806).
Please try to see why it's not working as intended, and fix it instead.

I'd keep the tests class instead.